src/app/user/make-observation/make-observation.component.ts

```diff
@@ -1,15 +1,72 @@
+import { HttpClient } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { GeolocationService } from '@ng-web-apis/geolocation';
 
 @Component({
   selector: 'app-make-observation',
   templateUrl: './make-observation.component.html',
-  styleUrls: ['./make-observation.component.scss']
+  styleUrls: ['./make-observation.component.scss'],
 })
 export class MakeObservationComponent implements OnInit {
+  public latitude;
+  public longitude;
 
-  constructor() { }
+  constructor(
+    private fb: FormBuilder,
+    private http: HttpClient,
+    private router: Router,
+    private readonly geolocation$: GeolocationService
+  ) {}
 
-  ngOnInit(): void {
+  obsForm = this.fb.group({
+    username: [''],
+    title: [''],
+    date: Date,
+    observation: [''],
+    city: [''],
+    state: [''],
+    longitude: Number,
+    latitude: Number,
+  });
+
+  ngOnInit(): void {}
+
+  getLocation() {
+    this.geolocation$.subscribe((position) => {
+      this.longitude = position.coords.longitude;
+      this.latitude = position.coords.latitude;
+    });
   }
 
+  async onSubmit() {
+    let obsData = {
+      user: localStorage.user,
+      title: this.obsForm.get('title').value,
+      text: this.obsForm.get('observation').value,
+      date: this.obsForm.get('date').value,
+      city: this.obsForm.get('city').value,
+      state: this.obsForm.get('state').value,
+      longitude: this.obsForm.get('longitude').value,
+      latitude: this.obsForm.get('latitude').value,
+    };
+    try {
+      const res: any = await this.http
+        .post('http://localhost:3000/observation/create', obsData)
```

Comment:
please use the environment.backendUrl father than hardcoding the URL here - this URL will only work locally and will break when we push to heroku